narwhals/utils.py

```diff
@@ -274,15 +274,37 @@ def maybe_get_index(obj: T) -> Any | None:
     return None
 
 
-def maybe_set_index(df: T, column_names: str | list[str]) -> T:
+def maybe_set_index(
+    df: T,
+    column_names: str | list[str] | None = None,
+    *,
+    index: Series | list[Series] | None = None,
+) -> T:
     """
-    Set columns `columns` to be the index of `df`, if `df` is pandas-like.
+    Set the index of `df`, if `df` is pandas-like, otherwise this is a no-op.
+
+    Arguments:
+        df: object for which maybe set the index (can be either a Narwhals `DataFrame`
+            or `Series`).
```

Comment:
is it a bit odd for this to be called `df` when it can be a dataframe or series?